tool/tctl/common/acl_command.go

```diff
@@ -0,0 +1,251 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/gravitational/trace"
+	"golang.org/x/exp/slices"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/api/types/accesslist"
+	"github.com/gravitational/teleport/lib/asciitable"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/service/servicecfg"
+	"github.com/gravitational/teleport/lib/utils"
+)
+
+// ACLCommand implements the `tctl acl` family of commands.
+type ACLCommand struct {
+	format string
+
+	ls          *kingpin.CmdClause
+	get         *kingpin.CmdClause
+	usersAdd    *kingpin.CmdClause
+	usersRemove *kingpin.CmdClause
+	usersList   *kingpin.CmdClause
+
+	// Used for managing a particular access list.
+	accessListName string
+
+	// Used for managing membership to an access list.
+	userName string
+	expires  string
+	reason   string
+}
+
+// Initialize allows ACLCommand to plug itself into the CLI parser
+func (c *ACLCommand) Initialize(app *kingpin.Application, _ *servicecfg.Config) {
+	acl := app.Command("acl", "Manage access lists.").Alias("access-lists")
+
+	c.ls = acl.Command("ls", "List cluster access lists.")
+	c.ls.Flag("format", "Output format, 'yaml', 'json', or 'text'").Default(teleport.YAML).EnumVar(&c.format, teleport.YAML, teleport.JSON, teleport.Text)
+
+	c.get = acl.Command("get", "Get detailed information for an access list.")
+	c.get.Arg("access-list-name", "The access list name.").Required().StringVar(&c.accessListName)
+	c.get.Flag("format", "Output format, 'yaml', 'json', or 'text'").Default(teleport.YAML).EnumVar(&c.format, teleport.YAML, teleport.JSON, teleport.Text)
+
+	users := acl.Command("users", "Manage user membership to access lists.")
+
+	c.usersAdd = users.Command("add", "Add a user to an access list.")
+	c.usersAdd.Arg("access-list-name", "The access list name.").Required().StringVar(&c.accessListName)
+	c.usersAdd.Arg("user", "The user to add to the access list.").Required().StringVar(&c.userName)
+	c.usersAdd.Arg("ttl", "When the user's access expires (must be in RFC3339).").Required().StringVar(&c.expires)
```

Comment:
Is this a TTL (which is a duration), or an absolute time?

If the latter, the flag should probably be called `--expires`

Reply:
That's correct, so I'll revert this back to expires.